### PR TITLE
pass meta to `defineProperty`

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: packages/ember-runtime/lib/system/core_object.js [hunk 1]
```diff
@@ -196,11 +196,7 @@ function makeCtor(base) {
             } else if (typeof self.setUnknownProperty === 'function' && !(keyName in self)) {
               self.setUnknownProperty(keyName, value);
             } else {
-              if (DEBUG) {
-                defineProperty(self, keyName, null, value); // setup mandatory setter
-              } else {
-                self[keyName] = value;
-              }
```
Comment (bekzod) [inline packages/ember-runtime/lib/system/core_object.js]: `defineProperty` defaults to same logic if descriptor argument is `null` as above

**Comment by rwjblue** [comment]
It definitely make sense to pass in the objects meta, but I’m not certain that we should change the production path to go through `defineProperty` here...

**Comment by bekzod** [comment]
internally `defineProperty` would lead to same code path as descriptor argument is null, won't complicate things :P so will just pass meta as you said :)

**Comment by bekzod** [comment]
internally `defineProperty` would lead to same code path as descriptor argument is null, won't complicate things :P so will just pass meta as you said :)

**Review by rwjblue — APPROVED** [review]

**Pull request merged**